app/vmui/packages/vmui/src/pages/ExploreLogs/hooks/useFetchLogs.ts

```diff
@@ -13,13 +13,21 @@ export const useFetchLogs = (server: string, query: string, limit: number) => {
 
   const url = useMemo(() => getLogsUrl(server), [server]);
 
+  const prepareQuery = (query: string) => {
+    const limitRegex = /\|\s*(limit|head)\s*:?[\s-]*\d+/gm;
```

Comment:
: is optional, you can have limit 10 or limit: 10

Reply:
and there will be more more things like that - so it worth to change logic of _time instead of add checks for limit

Reply:
yes, it works just like that, `:` is optional

Reply:
In any case, we need to wrap part of the expression in parentheses to ensure that the time range applies to all parts of the expression. Here is a more detailed explanation of why this is necessary: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5920.

At the moment, I don't understand what format such things will have, but perhaps we will be able to easily identify them and move them outside the brackets. If you have any ideas on how to change the logic, let me know.

Alternatively, you can pass the time range as parameters in the request and solve this problem on the server side.

Reply:
we are going to add `| fields *` or `| fields container_name, level` as next function, just keep in mind